File: client/src/main/java/au/com/codeka/warworlds/client/world/CompleteStarCollection.java
package au.com.codeka.warworlds.client.world;

import au.com.codeka.warworlds.client.game.world.StarCollection;
import au.com.codeka.warworlds.common.proto.Star;

/**
 * Implementation of {@link StarCollection} that represents all of your empire's stars (though we
 * do support filtering stars).
 */
public class CompleteStarCollection implements StarCollection {
  @Override
  public int size() {
    return 0;
  }

  @Override
  public Star get(int index) {
    return null;
  }

  @Override
  public int indexOf(long starId) {
    return 0;
  }

  @Override
  public boolean notifyStarModified(Star star) {
    return false;
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/world/CompleteStarCollection.kt
package au.com.codeka.warworlds.client.world

import au.com.codeka.warworlds.client.game.world.StarCollection
import au.com.codeka.warworlds.common.proto.Star

/**
 * Implementation of [StarCollection] that represents all of your empire's stars (though we
 * do support filtering stars).
 */
class CompleteStarCollection : StarCollection {
  override fun size(): Int {
    return 0
  }

  override fun get(index: Int): Star {
    throw NotImplementedError()
  }

  override fun indexOf(starId: Long): Int {
    return 0
  }

  override fun notifyStarModified(star: Star): Boolean {
    return false
  }
}